src/Features/LanguageServer/Protocol/ExternalAccess/Razor/GetProjectContextsHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Razor;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(GetProjectContextsHandler)), Shared]
+[Method(GetProjectContextsName)]
+internal class GetProjectContextsHandler : ILspServiceDocumentRequestHandler<ProjectContextsParams, ProjectContextList?>
+{
+    private const string GetProjectContextsName = "roslyn/getProjectContexts";
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public GetProjectContextsHandler()
+    {
+    }
+
+    public bool MutatesSolutionState => false;
+
+    public bool RequiresLSPSolution => true;
+
+    public TextDocumentIdentifier GetTextDocumentIdentifier(ProjectContextsParams request)
+        => request.TextDocument;
+
+    public Task<ProjectContextList?> HandleRequestAsync(ProjectContextsParams request, RequestContext context, CancellationToken cancellationToken)
+    {
+        Contract.ThrowIfNull(context.Workspace);
+        Contract.ThrowIfNull(context.Solution);
+
+        // We specifically don't use context.Document here because we want multiple
+        var documents = context.Solution.GetDocuments(request.TextDocument.Uri);
+
+        if (!documents.Any())
+        {
+            return SpecializedTasks.Null<ProjectContextList>();
+        }
+
+        var idToIntermediateOutputMap = new Dictionary<string, string?>();
+        var contexts = new List<VSProjectContext>();
+
+        foreach (var document in documents)
+        {
+            var project = document.Project;
+            var projectContext = ProtocolConversions.ProjectToProjectContext(project);
+            contexts.Add(projectContext);
+            idToIntermediateOutputMap[projectContext.Id] = project.CompilationOutputInfo.AssemblyPath;
```

Comment:
Strictly speaking this doesn't match the doc comment on the POCO, since this `AssemblyPath` is the path to the DLL, and an IntermediateOutputPath is the path to the folder that contains the DLL. I'm not fussed which bit you choose to change though :)